.circleci/config.yml

```diff
@@ -1,17 +1,145 @@
+          command: opsctl validate releases -p kvm.yaml
+      - run:
+          name: Validate AWS releases
+          command: opsctl validate releases -p aws.yaml
+      - run:
+          name: Validate Azure releases
+          command: opsctl validate releases -p azure.yaml
+  prepare-helm-chart:
+    docker:
+      - image: bash:5.0.16
+    parameters:
+      app_name:
+        description: "Name of helm chart."
+        type: "string"
+      source:
+        description: "Platform releases YAML to deploy as helm chart."
+        type: "string"
+    steps:
+      - checkout
+      - run:
+          name: Create chart directory
+          command: mkdir -p helm/<<parameters.app_name>>/templates
+      - run:
+          name: Copy template release into chart templates
+          command: mv .circleci/release.yaml helm/<<parameters.app_name>>/templates
+      - run:
+          name: Copy releases to values.yaml
+          command: |
+            echo "versions:" > helm/<<parameters.app_name>>/values.yaml
+            sed 's/^/  /' aws.yaml >> helm/<<parameters.app_name>>/values.yaml
+      - run: 
+          name: Create Chart.yaml
+          command: |
+            cat \<<EOF > helm/<<parameters.app_name>>/Chart.yaml
+            name: <<parameters.app_name>>
+            version: [[ .Version ]]
+            EOF
+      - persist_to_workspace:
+          root: .
+          paths:
+            - ./helm
+
+workflows:
+  workflow:
+    jobs:
+      - validate
+      # KVM
+      - prepare-helm-chart:
+          name: prepare-helm-chart-kvm
+          app_name: releases-kvm
+          requires:
+            - validate
+          source: kvm
+      - architect/push-to-app-catalog:
+          name: push-releases-kvm-to-control-plane-app-catalog
+          app_catalog: control-plane-catalog
+          app_catalog_test: control-plane-test-catalog
+          attach_workspace: true
+          chart: releases-kvm
+          requires:
+            - prepare-helm-chart-kvm
+      - architect/push-to-app-collection:
+          name: push-releases-to-kvm-app-collection
+          app_name: releases-kvm
+          app_collection_repo: kvm-app-collection
+          unique: "true"
+          requires:
+            - push-releases-kvm-to-control-plane-app-catalog
+          filters:
+            # Trigger the job on merge to master.
+            branches:
+              only: master
+      # AWS
+      - prepare-helm-chart:
+          name: prepare-helm-chart-aws
+          app_name: releases-aws
+          requires:
+            - validate
+          source: aws
+      - architect/push-to-app-catalog:
+          name: push-releases-aws-to-control-plane-app-catalog
+          app_catalog: control-plane-catalog
+          app_catalog_test: control-plane-test-catalog
+          attach_workspace: true
+          chart: releases-aws
+          requires:
+            - prepare-helm-chart-aws
+            - push-releases-kvm-to-control-plane-app-catalog
```

Comment:
This looks weird, but it's required to avoid a race when multiple `architect/push-to-app-catalog` jobs are committing to the same repo. They need to run in sequence.

Reply:
It actually makes sense to me - just the expected circleci dance.